collector/mdadm.go

```diff
@@ -0,0 +1,200 @@
+// +build !nomdadm
+
+package collector
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/log"
+)
+
+var statusfile = "/proc/mdstat"
+
+type mdstatus struct {
+	mdName       string
+	isActive     bool
+	disksActive  int64
+	disksTotal   int64
+	blocksTotal  int64
+	blocksSynced int64
+}
+
+type mdadmCollector struct {
+	isActive     *prometheus.GaugeVec
+	disksActive  *prometheus.GaugeVec
+	disksTotal   *prometheus.GaugeVec
+	blocksTotal  *prometheus.GaugeVec
+	blocksSynced *prometheus.GaugeVec
+}
+
+func init() {
+	Factories["mdadm"] = NewMdadmCollector
+}
+
+func evalStatusline(statusline string) (active, total, size int64) {
+
+	first_split := strings.Split(statusline, "[")
+	second_split := strings.Split(first_split[1], "]")
+
+	sizeDirty := strings.Split(first_split[0], "blocks")[0]
+	sizeStripped := strings.Replace(sizeDirty, " ", "", -1)
+
+	numberOfDrives := strings.Split(second_split[0], "/")
+
+	total, _ = strconv.ParseInt(numberOfDrives[0], 10, 64)
```

Comment:
I'd just return a meaningful error message (like `return fmt.Errorf("parsing number of drives failed: %s", err)`) all the way to `Update()` and from there it will be returned to the collector-handling framework, which will log already anyways. So I wouldn't log here at all.